### DISCUSSION TITLE
Help with Alignment Marks: recommendation?

### DISCUSSION THREAD

**Discussion by schneich** [jomjol/AI-on-the-edge-device#2414] [Q&A] [answered]
Hi,
I finally got my ESP-32-CAM running. Unfortunately, I am now facing shifted Alignment Marks.
I have found this in the documentation (https://jomjol.github.io/AI-on-the-edge-device-docs/Error-Debugging/#roi-misaligned), but I am still struggling, as I am missing the background know-how.

### **the error**
![grafik](https://github.com/jomjol/AI-on-the-edge-device/assets/84641760/34a0ec81-30bf-48fd-b9a2-9fe562a8e632)

### **Settings -> Reference Image**
![grafik](https://github.com/jomjol/AI-on-the-edge-device/assets/84641760/04ea3d3a-ba94-45d9-b5a5-7e3488aab65d)
(To reduce reflection from the LED, I reduced the intensity to 20 and have put a little bit of "hot glue" over the LED.)

### **Settings -> Alignment Marks: 0**
![grafik](https://github.com/jomjol/AI-on-the-edge-device/assets/84641760/3c01a780-1f43-4654-9d23-9403ffba479a)

### **Settings -> Alignment Marks: 1**
![grafik](https://github.com/jomjol/AI-on-the-edge-device/assets/84641760/c2b7e4f4-c12f-4564-b732-d10f712fa182)

### **Questions**
- In Settings -> Configuration (Expert Mode), there is the option to change Alignment Search Field X & Y. Does this has to be in line with the size of the Alignment Marks (see screenshot above)? ![grafik](https://github.com/jomjol/AI-on-the-edge-device/assets/84641760/a255c7bd-01af-4006-a796-c87cac0301f2)
- What is your recommendation for the Alignment Marks: better bigger or smaller? What would you tag in my screenshots?
- How do you handle the reflection on the _wheels_ of the numbers? (During first run, the figures were read correctly, but it is not ideal, I guess.)

Any help is very much appreciated,
Chris

**Comment by schneich** [comment, marked as answer]
I just created my own "Alignment Marks" and two runs have now been successful. 🎉
I will keep it running for a couple of days and see, what happens.
![grafik](https://github.com/jomjol/AI-on-the-edge-device/assets/84641760/3286d850-377d-483d-90ab-1cf9f680ed7a)


If anybody is interested: these are the examples, I have created. Feel free to test, use, change, improve them. (If someone of the tech guys could explain, how the algorithm really works, we might come up with a more efficient Alignment Mark?) 

[AI_on_the_Edge_AlignmentMarks.pdf](https://github.com/jomjol/AI-on-the-edge-device/files/11501958/AI_on_the_Edge_AlignmentMarks.pdf)
[AI_on_the_Edge_AlignmentMarks.pptx](https://github.com/jomjol/AI-on-the-edge-device/files/11501959/AI_on_the_Edge_AlignmentMarks.pptx)

Chris

**Comment by caco3** [comment]
I am surprised why your initial marks do not work. They look ok. Maybe it would help to make them a bit smaller.

This works fine for me:
![grafik](https://github.com/jomjol/AI-on-the-edge-device/assets/1783586/c62acdc4-de35-4daa-bab6-e0b38d82fe99)

IMO there is no need to change the Expert Parameters.

**Comment by schneich** [comment]
Hi @caco3,
thank you for your replay. Ok, I over-engineered my Alignment Marks a little bit. 😄 A simple black dot on a white sticker is enough?
Has the size of the Alignment Marks an influence on the computation time of each run?

Chris

**Reply by caco3** [reply]
> Has the size of the Alignment Marks an influence on the computation time of each run?

Not as I am aware off, but I am not sure.

**Comment by forrestfunk** [comment]
I don't think they are over-engineered. I use a similar kind of alignment marks. Works great for over 2 months now.
[alignment marks.pdf](https://github.com/jomjol/AI-on-the-edge-device/files/11522267/alignment.marks.pdf)
